blueflood-core/src/main/java/com/rackspacecloud/blueflood/cache/TokenCache.java

```diff
@@ -0,0 +1,75 @@
+package com.rackspacecloud.blueflood.cache;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.rackspacecloud.blueflood.types.Token;
+import com.rackspacecloud.blueflood.utils.Metrics;
+
+import java.util.concurrent.TimeUnit;
+
+public class TokenCache {
+
+    // this collection is used to reduce the number of tokens that get written.
+    // Simply, if a token has been seen within the last 10 minutes, don't bother.
```

Comment:
If we index a locator `1111:a.b.c.d` for tokens. we index documents with id's `1111:a, 1111:a.b, 1111:a.b.c, 1111:a.b.c.d:$` as tokens. Next time we get a locator`1111:a.b.c.e`, we only want to index `1111:a.b.c.e:$` as we had already indexed previous layers. Does this answer your question?